Library/Formula/sha2.rb

```diff
@@ -0,0 +1,20 @@
+class Sha2 < Formula
+  homepage "http://www.aarongifford.com/computers/sha.html"
+  url "http://www.aarongifford.com/computers/sha2-1.0.1.tgz"
+  sha1 "ffbc69e0b0ab47f70e61aeb14fa16ba2b74dc728"
+
+  option "without-check", "Skip compile-time tests"
+
+  def install
+    system ENV.cc, "-O2", "-DSHA2_UNROLL_TRANSFORM", "-Wall", "-o", "sha2", "sha2prog.c", "sha2.c"
```

Comment:
No `Makefile`, and the README says:

> * SHA2_UNROLL_TRANSFORM
> By defining this either on the command line or in sha2.h or sha2.c,
> the code will use macros to partially "unroll" the SHA transform
> function.  This usually generates bigger executables.  It CAN (but
> not necessarily WILL) generate faster code when you tell your compiler
> to optimize things.  For example, on the FreeBSD and Linux x86 systems
> I tested things on (using gcc), when I optimized with just -O2 and
> unrolled the transform, the hash transform was faster by 15-30%.  On
> these same systems, if I did NO optimization, the unrolled transform
> was SLOWER, much slower (I'm guessing because the code was breaking
> the cache, but I'm not sure).  Your mileage may vary.